mesh/model.c

```diff
@@ -815,40 +815,115 @@ static int add_sub(struct mesh_net *net, struct mesh_model *mod,
 	return MESH_STATUS_SUCCESS;
 }
 
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <errno.h>
+
+#define DEV_KEY_MSG 0
+#define APP_KEY_MSG 1
+
+struct msg {
+
+	uint8_t element;
+	uint8_t src;
+
+	uint8_t type;
+	union {
+		struct {
+			uint8_t net_idx;
+			bool remote;
+		} dev;
+
+		struct {
+			uint8_t app_idx;
+			uint16_t dst;
+			uint8_t label[16];
+		} app;
+	};
+
+	uint8_t *data;
+};
+
+static void send_msg_io(struct l_io *io, const uint8_t *value, size_t length)
+{
+	struct msghdr msg;
+	struct iovec iov;
+	int err;
+
+	iov.iov_base = (void *)value;
+	iov.iov_len = length;
+
+	memset(&msg, 0, sizeof(msg));
+	msg.msg_iov = &iov;
+	msg.msg_iovlen = 1;
+
+	err = sendmsg(l_io_get_fd(io), &msg, MSG_NOSIGNAL);
+	if (err < 0)
```

Comment:
I would just ignore errors, we don't want to spam logs if application is too slow.